aws/table_aws_ssm_inventory_entry.go

```diff
@@ -0,0 +1,151 @@
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+func tableAwsSSMInventoryEntry(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_ssm_inventory_entry",
+		Description: "AWS SSM Inventory Entry",
+		List: &plugin.ListConfig{
+			ParentHydrate: listAwsSSMInventories,
+			Hydrate:       listAwsSSMInventoryEntries,
+			KeyColumns: plugin.KeyColumnSlice{
+				{Name: "instance_id", Require: plugin.Optional},
+				{Name: "type_name", Require: plugin.Optional},
+			},
+		},
+		GetMatrixItemFunc: SupportedRegionMatrix(ssmv1.EndpointsID),
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "instance_id",
+				Description: "The managed node ID targeted by the request to query inventory information.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "type_name",
+				Description: "The type of inventory item returned by the request.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "capture_time",
+				Description: "The time that inventory information was collected for the managed node(s).",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "schema_version",
+				Description: "The inventory schema version used by the managed node(s).",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "entries",
+				Description: "The inventory items on the managed node(s).",
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("InstanceId"),
+			},
+		}),
+	}
+}
+
+type InventoryEntryInfo struct {
+	CaptureTime   *string
+	Entries       map[string]string
+	InstanceId    *string
+	SchemaVersion *string
+	TypeName      *string
+}
+
+//// LIST FUNCTION
+
+func listAwsSSMInventoryEntries(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	inventory := h.Item.(*InventoryInfo)
+
+	// Create session
+	svc, err := SSMClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_ssm_inventory_entries.listAwsSSMInventoryEntry", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region check
+		return nil, nil
+	}
+
+	if d.EqualsQualString("instance_id") != "" {
+		if d.EqualsQualString("instance_id") != *inventory.Id {
+			return nil, nil
+		}
+	}
+	if d.EqualsQualString("type_name") != "" {
+		if d.EqualsQualString("type_name") != *inventory.TypeName {
+			return nil, nil
+		}
+	}
+
+	maxItems := int32(50)
+
+	// Reduce the basic request limit down if the user has only requested a small number of rows
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxItems {
+			if limit < 1 {
+				maxItems = int32(1)
```

Comment:
Since the SDK already handles this condition, I think we can remove this check.

Reply:
Updated the logic